lineage/figures/figure12.py

```diff
@@ -0,0 +1,213 @@
+        _, obs1 = list(zip(*lineage.lineage_stats[1].full_lin_cells_obs))
+
+        p = sp.gamma.pdf(obs0, a=a0[i], loc=gamma_loc, scale=scale0[i])
+        q = sp.gamma.pdf(obs1, a=a0[i], loc=gamma_loc, scale=scale1[i])
+
+        size = min(p.shape[0], q.shape[0])
+        if size == 0:
+            raise ValueError('# of cells in one of the states is zero!')
+        else:
+            pprime = random.sample(list(p), size)
+            qprime = random.sample(list(q), size)
+            # find the KL divergence
+        gammaKL1.append(sp.entropy(np.asarray(pprime), np.asarray(qprime)))
+
+        X = [lineage]
+        states = [cell.state for cell in lineage.output_lineage]
+        num_iter = 5  # for every KL value, it runs the model 5 times
+        # accuracy and returns the avg accuracy for 5 iters
+        for j in range(num_iter):
+            _, _, all_states, tHMMobj, _, _ = Analyze(X, 2)
+
+            # find the accuracy
+            temp = accuracy(tHMMobj, all_states)[0] * 100
+
+            acc1.append(temp)
+    gammaKL_total.append(gammaKL1)
+
+    for j in range(10):
+        tmp = np.sum(acc1[j:num_iter * (j + 1)]) \
+            / len(acc1[j:num_iter * (j + 1)])
+        acc.append(tmp)
+    return acc, gammaKL_total[0]
+
+
+def distributionPlot():
+    """ Here we plot the distributions used in the KL divergence plot
+    to show how far the distributions are for each point.
+    For 10 distributions, 500 random variables are generated
+    and to be used by the violinplot they are concatenated in
+    "lifetime" column of the DataFrame, and to separate them,
+    there is another column named "distributions". """
+    gamma_loc = 0
+    bern_p0 = 0.99
+    bern_p1 = 0.88
+    a0 = np.linspace(5.0, 17.0, 10)
+    scale0 = 10 * ([2.0])
+    a1 = np.linspace(30.0, 17.0, 10)
+    scale1 = 10 * ([3.0])
+
+    dist1 = np.concatenate((sp.gamma.rvs(a=a0[9], loc=gamma_loc,
+                                         scale=scale0[9],
+                                         size=500),
+                            sp.gamma.rvs(a=a1[9], loc=gamma_loc,
+                                         scale=scale1[9],
+                                         size=500),
+                            sp.gamma.rvs(a=a0[8], loc=gamma_loc,
+                                         scale=scale0[8],
+                                         size=500),
+                            sp.gamma.rvs(a=a1[8],
+                                         loc=gamma_loc, scale=scale1[8],
+                                         size=500),
+                            sp.gamma.rvs(a=a0[7], loc=gamma_loc,
+                                         scale=scale0[7], size=500),
+                            sp.gamma.rvs(a=a1[7], loc=gamma_loc,
+                                         scale=scale1[7], size=500),
+                            sp.gamma.rvs(a=a0[6], loc=gamma_loc,
+                                         scale=scale0[6], size=500),
```

Comment:
Surely there's a loop you can use to do this?

Reply:
Oh, that's right. will do right away.
And I'm not using the function, I'm using the entropy, just forgot to remove it. Will do that, too.
Thanks.